### PR TITLE
Add bandit

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@MartinDelille sorry, it's `target_compile_features`: https://cmake.org/cmake/help/latest/command/target_compile_features.html

File: recipes/bandit/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+find_package(bandit REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} bandit::bandit)
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```
Comment (MartinDelille) [inline recipes/bandit/all/test_package/CMakeLists.txt:8]: @uilianries Shall I use `target_compile_features` here too ?

Comment (uilianries) [reply recipes/bandit/all/test_package/CMakeLists.txt:8]: @MartinDelille Yes, please.

File: recipes/bandit/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class BanditConan(ConanFile):
+    name = "bandit"
+    description = "Human-friendly unit testing for C++11"
+    topics = ("testing", "header-only")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/banditcpp/bandit"
+    license = "MIT"
+    no_copy_source = True
```
Comment (uilianries) [inline recipes/bandit/all/conanfile.py:16]:
```suggestion
```
As we need to apply a patch, let's avoid doing it over the source folder. For debugging and development, it's a mess for us.

File: recipes/bandit/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    def build(self):
+        apply_conandata_patches(self)
+
+    def package(self):
+        copy(self, "LICENSE.txt", src=os.path.join(self.source_folder, "docs"), dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, pattern="*", dst=os.path.join(self.package_folder, "include", "bandit"), src=os.path.join(self.source_folder, "bandit"))
+
+    def package_id(self):
+        self.info.clear()
```
Comment (uilianries) [inline recipes/bandit/all/conanfile.py:40]:
```suggestion
        self.info.clear()

    def package_info(self):
        self.cpp_info.bindirs = []
        self.cpp_info.frameworkdirs = []
        self.cpp_info.libdirs = []
        self.cpp_info.resdirs = []

```
As we don't use those folders, let's sanitize them.

File: recipes/bandit/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
```
Comment (uilianries) [inline recipes/bandit/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.8)
```
https://cmake.org/cmake/help/latest/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#high-level-meta-features-indicating-c-standard-support

File: recipes/bandit/all/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "VirtualRunEnv"
```
Comment (uilianries) [inline recipes/bandit/all/test_package/conanfile.py:9]:
```suggestion
    generators = "CMakeDeps", "VirtualRunEnv"
    test_type = "explicit"
```
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#changes-in-the-test-package-recipe

File: recipes/bandit/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
```
Comment (uilianries) [inline recipes/bandit/all/test_package/conanfile.py:17]: You can avoid it by declaring "CMakeToolchain" as generator, but using explicitly is more than welcome.

**Review by uilianries — COMMENTED** [review]
Almost good to go, only small details.

File: recipes/bandit/all/test_package/conanfile.py (hunk 6 above)
Comment (MartinDelille) [reply recipes/bandit/all/test_package/conanfile.py:17]: Ok so I leave it like that ?

**Review by danimtb — APPROVED** [review]

**Review by toge — APPROVED** [review]

File: recipes/bandit/all/test_package/conanfile.py (hunk 6 above)
Comment (uilianries) [reply recipes/bandit/all/test_package/conanfile.py:18]: @MartinDelille yes, you can leave generate with CMakeToolchain always.

**Pull request merged**